crates/edr_rpc_client/src/cache.rs

```diff
@@ -0,0 +1,133 @@
+/// Types for caching block specifications.
+pub mod block_spec;
+pub(crate) mod chain_id;
+/// Types for caching filters.
+pub mod filter;
+mod hasher;
+/// Types for indexing the cache.
+pub mod key;
+
+use std::{
+    fmt::Debug,
+    io,
+    path::{Path, PathBuf},
+    time::Instant,
+};
+
+use serde::de::DeserializeOwned;
+
+pub use self::hasher::KeyHasher;
+use self::key::{ReadCacheKey, WriteCacheKey};
+use crate::RpcClientError;
+
+/// Trait for RPC method types that can be cached.
+pub trait CacheableMethod: Sized {
```

Comment:
The trait `CacheableMethod` having `block_number_request` and `chain_id_request` methods is a bit confusing as `eth_blockNumber` and `eth_chainId` are not cacheable.

Reply:
Do you have an idea for a more applicable trait name?

The flow is as follows: 
- the method type enum that's passed into the `RpcClient` should be cacheable
- our caching solution needs to know the block number and chain ID, hence the existence of those two functions.
- a subset of the method type enum can actually be cached. This is called a `CachedMethod`

Reply:
One alternative name I can imagine is to
- rename `CacheableMethod` to `RpcMethod` and `CachedMethod` to `CacheableMethod`

Would that clarify things?

Reply:
I tried to clarify things in https://github.com/NomicFoundation/edr/pull/483/commits/b34068e14bf6e8e631106e5f412530a0a6b26944

Reply:
Thanks the change makes sense!